subsys/mgmt/mcumgr/grp/os_mgmt/src/os_mgmt.c

```diff
@@ -499,7 +502,9 @@ os_mgmt_bootloader_info(struct smp_streamer *ctxt)
 		ok = ok && zcbor_tstr_put_lit(zse, "no-downgrade") &&
 		     zcbor_bool_encode(zse, &(bool){true});
 #endif
-	} else {
+	} else
+#endif
+	{
```

Comment:
struggling real hard with this, how about

```c
#if !defined(CONFIG_BOOTLOADER_MCUBOOT)
        return OS_MGMT_ERR_QUERY_YIELDS_NO_ANSWER;
#else
        if (decoded == 0) {
                ok = zcbor_tstr_put_lit(zse, "bootloader") &&
                     zcbor_tstr_put_lit(zse, "MCUboot");
        } else if (zcbor_map_decode_bulk_key_found(bootloader_info, ARRAY_SIZE(bootloader_info),
                   "query") &&
                   (sizeof("mode") - 1) == query.len &&
                   memcmp("mode", query.value, query.len) == 0) {

                ok = zcbor_tstr_put_lit(zse, "mode") &&
                     zcbor_int32_put(zse, BOOTLOADER_MODE);
#ifdef CONFIG_MCUBOOT_BOOTLOADER_NO_DOWNGRADE
                ok = ok && zcbor_tstr_put_lit(zse, "no-downgrade") &&
                     zcbor_bool_encode(zse, &(bool){true});
#endif
        } else {
                return OS_MGMT_ERR_QUERY_YIELDS_NO_ANSWER;
        }
#endif
```

Reply:
maybe an extra function would help, like having the "else" block there separate

Reply:
The reason I went for it like this is that it allows another to be added like this:
```
        /* If no parameter is recognized then just introduce the bootloader. */
#if defined(CONFIG_BOOTLOADER_MCUBOOT)
        if (decoded == 0) {
                ok = zcbor_tstr_put_lit(zse, "bootloader") &&
                     zcbor_tstr_put_lit(zse, "MCUboot");
        } else if (zcbor_map_decode_bulk_key_found(bootloader_info, ARRAY_SIZE(bootloader_info),
                   "query") &&
                   (sizeof("mode") - 1) == query.len &&
                   memcmp("mode", query.value, query.len) == 0) {

                ok = zcbor_tstr_put_lit(zse, "mode") &&
                     zcbor_int32_put(zse, BOOTLOADER_MODE);
#ifdef CONFIG_MCUBOOT_BOOTLOADER_NO_DOWNGRADE
                ok = ok && zcbor_tstr_put_lit(zse, "no-downgrade") &&
                     zcbor_bool_encode(zse, &(bool){true});
#endif
        } else
#endif
#if defined(CONFIG_BOOTLOADER_TWIDDLE)
        if (decoded == 0) {
                ok = zcbor_tstr_put_lit(zse, "bootloader") &&
                     zcbor_tstr_put_lit(zse, "twittle");
        } else if (zcbor_map_decode_bulk_key_found(bootloader_info, ARRAY_SIZE(bootloader_info),
                   "query") &&
                   (sizeof("mode") - 1) == query.len &&
                   memcmp("mode", query.value, query.len) == 0) {

                ok = zcbor_tstr_put_lit(zse, "mode") &&
                     zcbor_int32_put(zse, something_here);
        } else
#endif
        {    
            ok = smp_add_cmd_err(zse, MGMT_GROUP_ID_OS, OS_MGMT_ERR_QUERY_YIELDS_NO_ANSWER);
        }
```
whereas the suggested one does not. I could switch it to having a bool which gets set if something is added and if not set does what the `{ ... }` does if that works better for you?